operator/cmd/mesh/install.go

```diff
@@ -347,7 +347,14 @@ func getProfileNSAndEnabledComponents(iop *v1alpha12.IstioOperator) (string, str
 	}
 
 	if configuredNamespace := v1alpha12.Namespace(iop.Spec); configuredNamespace != "" {
-		return iop.Spec.Profile, configuredNamespace, enabledComponents, nil
+		return renderWithDefault(iop.Spec.Profile, "default"), configuredNamespace, enabledComponents, nil
```

Comment:
Should we just be setting the iop.Spec.Profile value somewhere? Not sure if we are already providing defaults directly to the spec in other parts of the code or not

Reply:
@howardjohn It seems like when flags are setting like "profile=demo", `overlaySetFlagValues` will set the path like {spec.profile: "demo"}. However, if none profile flag is provided, the profile field will be empty. I've revised and PTAL.